packages/caravan-fees/src/rbf.ts

```diff
@@ -0,0 +1,459 @@
+ * @see https://bitcoinops.org/en/newsletters/2023/10/25/#fn:rbf-warning Bitcoin Optech on replacement cycle attacks
+ */
+export const createAcceleratedRbfTransaction = (
+  options: AcceleratedRbfOptions,
+): string => {
+  const {
+    fullRBF = false,
+    strict = false,
+    originalTx,
+    network,
+    targetFeeRate,
+    absoluteFee,
+    availableInputs,
+    requiredSigners,
+    totalSigners,
+    scriptType,
+    changeIndex,
+    changeAddress,
+    reuseAllInputs = true,
+  } = options;
+
+  // Step 1: Validate change output options
+  if (changeIndex !== undefined && changeAddress !== undefined) {
+    throw new Error(
+      "Provide either changeIndex or changeAddress, not both. This ensures unambiguous handling of the change output.",
+    );
+  }
+  if (changeIndex === undefined && changeAddress === undefined) {
+    throw new Error(
+      "Either changeIndex or changeAddress must be provided for handling the change output.",
+    );
+  }
+
+  // Step 2: Analyze the original transaction
+  const txAnalyzer = new TransactionAnalyzer({
+    txHex: originalTx,
+    network,
+    targetFeeRate,
+    absoluteFee,
+    availableUtxos: availableInputs,
+    requiredSigners,
+    totalSigners,
+    addressType: scriptType,
+    changeOutputIndex: changeIndex,
+  });
+
+  txAnalyzer.assumeRBF = fullRBF;
+
+  // Step 3: Validate RBF possibility and suitability
+  validateRbfPossibility(txAnalyzer, fullRBF, strict);
+
+  // Step 4: Create a new transaction template
+  // Step 5: Add inputs from the original transaction
+  const { newTxTemplate, addedInputs } = addOriginalInputs(
+    txAnalyzer,
+    availableInputs,
+    { ...options, reuseAllInputs },
+    true, // This is an accelerated transaction
+  );
+
+  // Step 6: Add all non-change outputs from the original transaction
+  txAnalyzer.getOutputTemplates().forEach((outputTemplate) => {
+    if (!outputTemplate.isMalleable) {
+      newTxTemplate.addOutput(
+        new BtcTxOutputTemplate({
+          address: outputTemplate.address,
+          amountSats: outputTemplate.amountSats,
+          locked: true,
+        }),
+      );
+    }
+  });
+
+  // Step 7: Add additional inputs if necessary
+  addAdditionalInputs(newTxTemplate, availableInputs, addedInputs, txAnalyzer);
+
+  // Step 8: Add or adjust change output
+  if (newTxTemplate.needsChangeOutput) {
+    const totalInputAmount = new BigNumber(newTxTemplate.totalInputAmount);
+    const totalOutputAmount = new BigNumber(newTxTemplate.totalOutputAmount);
+    const fee = BigNumber.max(
+      newTxTemplate.targetFeesToPay,
+      txAnalyzer.minimumRBFFee,
+    );
+    const changeAmount = totalInputAmount.minus(totalOutputAmount).minus(fee);
+
+    const changeOutput = new BtcTxOutputTemplate({
+      address: changeAddress || txAnalyzer.outputs[changeIndex!].address,
+      amountSats: changeAmount.toString(),
+      locked: false,
+    });
```

Comment:
this can be a future optimization but I think this wouldn't be necessary if you just added a 0 amount output for change and then used `adjustChangeOutput`. The one thing you are doing extra here is setting fee to max of target fees and minimum RBF, but if you initialize the newTxTemplate to have target fee rate of `BigNumber.max(newTxTemplate.targetFeesToPay, txAnalyzer.minimumRBFFee);` then adjustChangeOutput should handle that just fine.

Reply:
changes done : [FINAL NITS:2](https://github.com/caravan-bitcoin/caravan/pull/114/commits/d46fa324190ac4304969351b1c6a9b44ba6d673e)